packages/firestore/gulpfile.js

```diff
@@ -36,6 +39,40 @@ const setupWatcher = () => {
   gulp.watch('src/**/*', buildModule);
 };
 
+/**
+ * This function is used to build `console.ts` see the file for the rationale
+ * behind its existence
+ */
+function buildConsole() {
+  return gulp
+    .src('console.ts')
+    .pipe(gulpWebpack(require('./webpack.config'), webpack))
+    .pipe(
+      through.obj(function(file, encoding, cb) {
+        if (file.isBuffer()) {
+          file.contents = Buffer.concat([
```

Comment:
Turns out the export name is super important. With any other name, jscompiler renames several important things (when in super-optimized mode).

I suspect we are pulling in this TS definition through some transitive dependency, and it prevents renaming:

```ts
declare let __firestore_exports__: {[key: string]: {}};
```

The following wrapper will get it all working, until we can gradually remove some of the incredible layers of jank 🤠:

```js
goog.module('firestore');
exports =
    (function() {
      var sdk_exports = eval(
          'var __firestore_exports__ = { firestore: {}};' +
          '__firestore_exports__.firestore=**CONTENT_HERE**\n' + 
          '__firestore_exports__');
      return sdk_exports.firestore;
    }).call(window);
```